src/Ethereum/Barz.cpp

```diff
@@ -85,4 +86,126 @@ Data getFormattedSignature(const Data& signature, const Data challenge, const Da
     return {};
 }
 
+Data getPrefixedMsgHash(const Data msgHash, const std::string& barzAddress, const uint32_t chainId) {
+    // keccak256("EIP712Domain(uint256 chainId,address verifyingContract)");
+    const Data& domainSeparatorTypeHashData = parse_hex("0x47e79534a245952e8b16893a336b85a3d9ea9fa8c573f3d803afb92a79469218");
```

Comment:
The `typedHash` itself is always static because it only includes the types.
As available in the implementation and unit test, the logic handles the final message hash if the `chainId` changes. (We have test added for `chainId` 3604 and 0)